scanpy/plotting/_tools/__init__.py

```diff
@@ -382,26 +389,41 @@ def _rank_genes_groups_plot(
         groupby = str(adata.uns[key]['params']['groupby'])
     group_names = adata.uns[key]['names'].dtype.names if groups is None else groups
 
-    var_names = {}  # dict in which each group is the key and the n_genes are the values
-    gene_names = []
-    for group in group_names:
-        if min_logfoldchange is not None:
-            df = rank_genes_groups_df(adata, group, key=key)
-            # select genes with given log_fold change
-            genes_list = df[df.logfoldchanges > min_logfoldchange].names.tolist()[
-                :n_genes
-            ]
+    if var_names is not None:
+        if isinstance(var_names, Mapping):
+            # get a single list of all gene names in the dictionary
+            var_names_list = sum([list(x) for x in var_names.values()], [])
+        elif isinstance(var_names, str):
+            var_names_list = [var_names]
         else:
-            # get all genes that are 'non-nan'
-            genes_list = [
-                gene for gene in adata.uns[key]['names'][group] if not pd.isnull(gene)
-            ][:n_genes]
+            var_names_list = var_names
+    else:
+        # dict in which each group is the key and the n_genes are the values
+        var_names = {}
+        var_names_list = []
+        for group in group_names:
+            df = rank_genes_groups_df(
+                adata,
+                group,
+                key=key,
+                gene_symbols=gene_symbols,
+                log2fc_min=min_logfoldchange,
```

Comment:
Would be great to homogenise the parameters. Here we have `log2fc_min` and `min_logfoldchange`. Do we have a strategy to make some parameters obsolete?

Also, would be good to allow the same filtering in the plots as `sc.get.rank_genes_groups_df`, namely `log2fc_min`, `log2fc_max` and `pval_cutoff`.

Reply:
I think we can leave this as an open question.